components/URLs.js

```diff
@@ -43,17 +43,24 @@ const StyleFixIconButton = styled.div`
   }
 `
 
-// Based on https://github.com/citizenlab/test-lists/blob/fd9620f6402f66f7835a831fdcd0731b449e9c52/scripts/lint-lists.py#L18
-const urlValidityRegex = /^(?:http)s?:\/\/(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})(?::\d+)?(?:\/?|[/?]\S+)$/i
-
 const validationSchema = Yup.object().shape({
   urls: Yup.array().of(
       Yup.object().shape({
           url: Yup.string()
               .required('cannot be empty')
               .matches(/^http(s?)/, 'should start with "https" or "http"')
-              .matches(urlValidityRegex, 'should be a valid URL format e.g "https://ooni.org/post/"')
-              .matches(/\/$/, 'should end with a slash e.g "https://ooni.org/"')
+              .test(
+                'is-valid-url',
+                'should be a valid URL format e.g "https://ooni.org/post/"',
+                (value) => {
+                  try {
+                    new URL(value)
```

Comment:
The URL parser seems to be a bit too permissive in what it accepts as a valid URL.

For example:

```
> new URL("https:/google.com")
> new URL("https:google.com")
> new URL("ftp://google.com")
> new URL("https://123.com")
```

Are all considered valid URLs by the URL parsing library, yet they are not going to lead to successful OONI measurements.

I suppose we could workaround the first two cases by automatically converting those URLs into valid ones (this can be done by calling `url.toString()`). This has the additional benefit that it will convert the URLs into puny code when entered via UTF-8 characters, that will not work in OONI Probe apps.
```
> (new URL("https://なし.com")).toString()
'https://xn--68jub.com/'
```

I think we should add the additional constraint that the url is using the `http` or `https` protocol.

According to [rfc1035](https://datatracker.ietf.org/doc/html/rfc1035#section-2.3.1) a domain name must start with a letter. We can also add this additional constraint to check if the first character in the `domain` attribute of the parsed URL is in fact a letter.